### PR TITLE
resolve template fields init checks for `bigquery`

### PR DESCRIPTION
**Pull request by okirialbert**
---

**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by okirialbert** [comment]
Resolve BigQuery templated fields logic checks #36484

File: airflow/providers/google/cloud/operators/bigquery.py [hunk 1]
```diff
@@ -2738,6 +2744,7 @@ def __init__(
         result_timeout: float | None = None,
         deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False),
         poll_interval: float = 4.0,
+        hook: BigQueryHook | None = None,
```
Comment (Taragolis) [inline airflow/providers/google/cloud/operators/bigquery.py]: Hook into initialiser for the different purpose. It setting into the execute method.
Better to use @cached_propery for this case. But anyway this change non relevant to the purpose of the PR 

```suggestion
```

File: airflow/providers/google/cloud/operators/bigquery.py [hunk 2]
```diff
@@ -2752,7 +2759,7 @@ def __init__(
         self.cancel_on_kill = cancel_on_kill
         self.result_retry = result_retry
         self.result_timeout = result_timeout
-        self.hook: BigQueryHook | None = None
+        self.hook = hook
```
Comment (Taragolis) [inline airflow/providers/google/cloud/operators/bigquery.py]:
```suggestion
        self.hook: BigQueryHook | None = None
```

File: airflow/providers/google/cloud/operators/bigquery.py (hunk 1 above)
Comment (okirialbert) [reply airflow/providers/google/cloud/operators/bigquery.py]: Noted. I've made the changes.

**Review by potiuk — APPROVED** [review]

**Pull request merged**